src/ngraph/op/fused/squeeze.cpp

```diff
@@ -73,24 +81,27 @@ void op::Squeeze::pre_validate_and_infer_types()
         set<size_t, greater<size_t>> unique_axes(begin(axes), end(axes));
         for (uint64_t axis : unique_axes)
         {
-            NODE_VALIDATION_CHECK(
-                this,
-                (data_shape.at(axis) == 1),
-                "provided axis value is invalid. Only axes of size 1 may be removed.");
+            if (!dynamic_data_shape)
+            {
+                auto data_shape = data.get_shape();
+                NODE_VALIDATION_CHECK(
+                        this,
+                        (data_shape.at(axis) == 1),
+                        "provided axis value is invalid. Only axes of size 1 may be removed.");
+            }
             axes_to_squeeze.at(axis) = 1;
         }
     }
 
-    Shape output_data_shape;
-    for (size_t idx = 0; idx < data_shape.size(); ++idx)
+    std::vector<Dimension> output_data_shape;
+    for (size_t idx = 0; idx < data_rank; ++idx)
```

Comment:
This `size_t` will probably need to be `uint64_t` when you make the other change.

Reply:
Thanks! Done